src/ngraph/runtime/gpu/cudnn_emitter.cpp

```diff
@@ -943,16 +943,27 @@ size_t runtime::gpu::CUDNNEmitter::build_convolution(const std::string& dtype,
         padding_below, window_movement_strides, window_dilation_strides, mode, data_type);
 
     cudnnConvolutionFwdAlgo_t conv_fwd_algo = CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_GEMM;
-    if (find_algo)
+    if (find_algo != algo_search::NONE)
     {
-        CUDNN_SAFE_CALL(cudnnGetConvolutionForwardAlgorithm(*m_ctx->cudnn_handle,
-                                                            tensor_desc_0,
-                                                            filter_desc,
-                                                            conv_desc,
-                                                            tensor_desc_1,
-                                                            CUDNN_CONVOLUTION_FWD_PREFER_FASTEST,
-                                                            /*memoryLimitInBytes=*/0,
-                                                            &conv_fwd_algo));
+        int num_algos;
+        int max_algos = 0;
+        CUDNN_SAFE_CALL(
+            cudnnGetConvolutionForwardAlgorithmMaxCount(*m_ctx->cudnn_handle, &max_algos));
+        std::vector<cudnnConvolutionFwdAlgoPerf_t> results(max_algos);
+        auto cudnn_algo_search = (find_algo == algo_search::EXPLICIT)
```

Comment:
just use `cudnnFindConvolutionForwardAlgorithm`

Reply:
same for others.

Reply:
In case we want to switch back, I left the algo_search enum, but set it to do the explicit search via `cudnnFindConvolutionForwardAlgorithm` by default.

Reply:
yes, keep it.